# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1436967864 14400
#      Wed Jul 15 09:44:24 2015 -0400
# Node ID e6af03353aceb54cd11558a4bfad50c802a3933f
# Parent  83d08eb76182561879667a368ab74598457e2aed
clean generated files in doc/interpreter directory

* doc/module.mk (doc-clean): Rename from clean-local.
(doc-maintainer-clean): Rename from maintainer-clean-local.
(interpreter-maintainer-clean): New target.
(doc-maintainer-clean-local): Depend on it.

* Makefile.am (clean-local, maintainer-clean-local): New targets.

diff --git a/doc/module.mk b/doc/module.mk
--- a/doc/module.mk
+++ b/doc/module.mk
@@ -365,16 +365,19 @@ EXTRA_DIST += \
   $(DOC_IMAGES) \
   $(DOC_IMAGES_SRC) \
   $(LOGOS) \
   $(TXI_SRC)
 
 interpreter-clean:
 	rm -rf t2d_cache
 
+interpreter-maintainer-clean:
+	rm -f $(BUILT_OCTAVE_TEXI_SRC)
+
 DISTCLEANFILES += \
   $(BUILT_TEXINFOS) \
   $(OCTAVE_HTML_STAMP)
 
 MAINTAINERCLEANFILES += \
   $(BUILT_DOC_IMAGES) \
   doc/interpreter/doc-cache
 
@@ -481,19 +484,19 @@ CLEANFILES += \
   doc/refcard/refcard-legal.log \
   doc/refcard/refcard-letter.log
 
 MAINTAINERCLEANFILES += \
   $(refcard_FORMATTED)
 
 DIRSTAMP_FILES += doc/refcard/$(octave_dirstamp)
 
-clean-local: interpreter-clean
+doc-clean: interpreter-clean
 
-maintainer-clean-local: doxyhtml-maintainer-clean
+doc-maintainer-clean: doxyhtml-maintainer-clean interpreter-maintainer-clean
 
 $(srcdir)/doc/interpreter/images.mk: $(srcdir)/doc/interpreter/config-images.sh $(srcdir)/doc/interpreter/images.awk $(srcdir)/doc/interpreter/images
 	$(srcdir)/doc/interpreter/config-images.sh $(top_srcdir)
 
 $(refcard_DVI) : %.dvi : %.tex doc/refcard/$(octave_dirstamp)
 	-$(AM_V_TEX)cd $(@D) && \
 	TEXINPUTS="$(abs_top_srcdir)/doc/refcard:$(TEXINPUTS):" \
 	$(TEX) $(<F) $(AM_V_texidevnull)
